src/include/duckdb/common/vector_operations/generic_executor.hpp

```diff
@@ -31,18 +31,21 @@ struct PrimitiveType {
 	}
 
 	INPUT_TYPE val;
+	bool is_null;
 
 	using STRUCT_STATE = PrimitiveTypeState;
 
-	static bool ConstructType(STRUCT_STATE &state, idx_t i, PrimitiveType<INPUT_TYPE> &result) {
+	static bool ConstructType(STRUCT_STATE &state, idx_t i, bool is_null, PrimitiveType<INPUT_TYPE> &result) {
 		auto &vdata = state.main_data;
 		auto idx = vdata.sel->get_index(i);
 		auto ptr = UnifiedVectorFormat::GetData<INPUT_TYPE>(vdata);
 		result.val = ptr[idx];
+		result.is_null = is_null;
 		return true;
 	}
 
 	static void AssignResult(Vector &result, idx_t i, PrimitiveType<INPUT_TYPE> value) {
+		FlatVector::SetNull(result, i, value.is_null);
```

Comment:
We only need to call `FlatVector::SetNull` when the value is null, so this could be:
```cpp
if (value.is_null) {
    FlatVector::SetNull(result, i, true);
}
```